WIP windsor body case

greole opened the pull request:
This PR adds the windsor body case, case folder structure, and basic automation.

greole committed:
initial commit windsor body

Co-authored-by: Loufli <louis.fliessbach@upstream-cfd.com>

greole committed:
initial commit windsor body

Co-authored-by: Loufli <louis.fliessbach@upstream-cfd.com>

greole commented on .gitignore:
```diff
@@ -0,0 +1,82 @@
+# Top-level build directories
+/build/
+/platforms/
+
```
```suggestion
__pycache__/
```

HendriceH replied:
Done

greole commented on WindsorBody/Allclean.pre:
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+rm -r WindsorBody_OF-v2206_incomp_AutoCFD3g1_test01 slurm-*.out images/*.png images/*.pdf
```
In general I would keep the folder names a bit more generic, thus WindrosBody_OF .. wont be in the file structure any more.

greole commented on WindsorBody/Allrun.pre:
```diff
@@ -0,0 +1,84 @@
+runDir="WindsorBody_OF-v2206_incomp_AutoCFD3g1_DDES"
+
+### Load modules as required
+module load gnu/11
```
The module structure is different on different machines. I would say we keep a separate folder for things that are machine specific.

greole commented on WindsorBody/Allrun.pre:
```diff
@@ -0,0 +1,84 @@
+else
+    srcDir="WindsorBody_OF-v2206_incomp_AutoCFD3g1.orig"
+fi
+runDir="WindsorBody_OF-v2206_incomp_AutoCFD3g1_DDES"
```
Again I probably would call the case just case and keep it in a WindsorBody folder.

greole commented on WindsorBody/Allrun.pre:
```diff
@@ -0,0 +1,84 @@
+   cp $use_control system/controlDict
+   mkdir -p $logDir
+   cd constant
+   tar -xvzf polyMesh.WindsorBody_AutoCFD3_c1g1.tar.gz > ../$logDir/01_unpackMesh.log 2>&1 || exit 1
```
I added a get_mesh script to pull the mesh from a google drive.

greole commented on WindsorBody/Scripts/python-files/IOFoamLib.py:
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
```
This needs some short documentation what it is, what it does, and how to use it.

greole commented on WindsorBody/case/system/decomposeParDict:
```diff
@@ -0,0 +1,40 @@
+
+// note: kahip decomposition can not be used in conjunction with new wall
+//       distance method "exactDistance" use hierachical instead
+method              kahip; //hierarchical;
```
Maybe we should use scotch since it works out of the box?

greole commented on WindsorBody/case/system/fvSolution.DDES:
```diff
@@ -0,0 +1,94 @@
+    "p|pcorr"
+    {
+        solver          GAMG;
+        tolerance       1e-99;
```
any reason for not using 0?

greole reviewed (commented):
First round of comments.

greole committed:
add an assets folder

greole committed:
add an assets folder

HendriceH committed:
Restructuring of case preparation script, deletion of unused files etc.

HendriceH committed:
Restructuring of case preparation script, deletion of unused files etc.

HendriceH commented:
@greole could you compress and upload the directory with the restart to the google drive as well, I set the workflow up to have it downloaded together with the mesh.
@Loufli we need to clean / adjust the Allrun scripts to work for generic hardware configurations. We can discuss this on Wedness day in the meeting about Automation

greole commented:
> @greole could you compress and upload the directory with the restart to the google drive as well, I set the workflow up to have it downloaded together with the mesh.

I'll do that, but I think it might make sense to keep the restart file in the assets folder. That way one can 

1.  run the simulation from zero to the restart file and validate the results 
2.  link, move or copy the restart into the case when needed and run from there for the benchmark runs

HendriceH commented:
> > @greole could you compress and upload the directory with the restart to the google drive as well, I set the workflow up to have it downloaded together with the mesh.
> 
> I'll do that, but I think it might make sense to keep the restart file in the assets folder. That way one can
> 
> 1. run the simulation from zero to the restart file and validate the results
> 2. link, move or copy the restart into the case when needed and run from there for the benchmark runs

Ah I think you missunderstood, I partially meant it how you describe it: The restart should not be bundled with the mesh, but separatly. So you can optionally download it, if desired. It might still need to go to google drive, as it might be too big for git.

greole committed:
add example OBR yml

greole committed:
add example OBR yml

greole committed:
fixup yml

greole committed:
fixup yml

Loufli committed:
Add reference data and update post processing

- Add reference data from Varney et al. read by post processing scripts
- Correct Allrun.pre
- Add description to python scripts
- Minor changes in python scripts

Loufli committed:
Add reference data and update post processing

- Add reference data from Varney et al. read by post processing scripts
- Correct Allrun.pre
- Add description to python scripts
- Minor changes in python scripts

Loufli committed:
Add bash workflow to concatenate case domain

- add bash script duplicateDomain.sh
- add OpenFOAM dictionaries (mapFieldsDict.*, createPatchDict.deleteIntern) for necessary mesh and field operations
- update Allrun.pre script

greole committed:
Merge branch 'case_windsor_body' of github.com:exasim-project/micro_benchmarks into case_windsor_body

greole committed:
add refineMeshScaling example

Loufli committed:
Delete unnecessary directory ._polyMesh and clean duplicateDomain.sh

Loufli committed:
Merge branch 'case_windsor_body' of github.com:exasim-project/micro_benchmarks into case_windsor_body

greole committed:
Fixup list syntax

greole committed:
update syntax for decomposition

greole committed:
add a first plot example

greole committed:
update plot script

greole committed:
add get_mesh script

Loufli committed:
Add paraview macros and update domain concatenation script

- add paraview macros in WindsorBody/assets/scripts/PV-Files/
- rename and restructure WindsorBody/duplicateDomain.sh in WindsorBody/assets/scripts/concatenateDomain.sh
- move WindsorBody/scripts/python-files/ to WindsorBody/assets/scripts/python-files/

Loufli committed:
Merge branch 'case_windsor_body' into case_windsor_body_testing

Loufli committed:
Add paraview macros and update domain concatenation script

- add paraview macros in WindsorBody/assets/scripts/PV-Files/
- rename and restructure WindsorBody/duplicateDomain.sh in WindsorBody/assets/scripts/concatenateDomain.sh
- move WindsorBody/scripts/python-files/ to WindsorBody/assets/scripts/python-files/

greole committed:
Merge branch 'case_windsor_body' of github.com:exasim-project/micro_benchmarks into case_windsor_body

greole committed:
add yaml.location

greole committed:
update yaml file

greole committed:
dont run checkMesh and make concat default

greole committed:
some fixes

greole committed:
Add postProcessLogs script

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
play around with plotting

greole committed:
update plot scripts

greole committed:
update plot scripts

greole committed:
update plot scripts

greole committed:
update plot scripts

greole committed:
update plot scripts

greole committed:
update plot scripts

greole committed:
update plot scripts

greole committed:
update plot scripts

greole committed:
update plot scripts

Loufli committed:
Merge branch 'case_windsor_body' into case_windsor_body_testing

Loufli committed:
Move OBR specific changes of setup to a OBR specific files and setup folder

Loufli committed:
Update .gitignore and WindsorBody/Allrun.pre

Loufli committed:
Merge branch 'case_windsor_body_testing' into case_windsor_body

Loufli committed:
Change OF version, turbulence model and solver settings

- OF v2206 to v2212
- SST-DDES to SA-DDES

Loufli committed:
Change turbulence model and restructure basicSetupOBR

- switch to Spalart-Allmaras as default turbulence model
- restructure OF output definition files
- add getFlowfield.sh script to download initial field
- add obr post-processing scripts
- change link in getMesh.sh

Loufli committed:
Change link to initial field

greole commented on LidDrivenCavity3D/assets/plot.py:
Leave out for later PR

greole commented on LidDrivenCavity3D/assets/scaling.yaml:
Check if that is the latest version see plots branch. Use scaling.yaml with includes.

greole commented on WindsorBody/Allclean.pre:
leave out of merge branch

greole commented on WindsorBody/Allrun.pre:
leave out of merge branch

greole commented on WindsorBody/assets/images/breakdown.png:
That is part of the data repository

greole commented on WindsorBody/assets/images/solve.png:
remove

greole commented on WindsorBody/assets/images/time_solve.png:
remove

greole commented on WindsorBody/assets/postProcessLogs.py:
data repository?

greole commented on WindsorBody/assets/scripts/PV-Files/PV-Macros/WindsorBody-Ensight_Slices.py:
Leave out of the merge branch

greole commented on WindsorBody/assets/scripts/PV-Files/PV-Macros/WindsorBody-Ensight_iso-Q-Cf.py:
Leave out of the merge branch

greole commented on WindsorBody/assets/scripts/PV-Files/PV-Macros/colouring.json:
Leave out of the merge branch

greole commented on WindsorBody/assets/scripts/PV-Files/PV-Macros/pvScriptLib.py:
Leave out of the merge branch

greole commented on WindsorBody/assets/scripts/concatenateDomain.sh:
keep

greole commented on WindsorBody/assets/scripts/concatenateDomainOBR.sh:
check merge conflicts, what is needed?

greole commented on WindsorBody/assets/scripts/getFlowfield.sh:
keep this.

greole commented on WindsorBody/assets/scripts/python-files/IOFoamLib.py:
leave on this branch, but keep out merge

greole commented on WindsorBody/assets/scripts/python-files/plotScript.py:
leave on this branch, but keep out merge

greole commented on WindsorBody/assets/test_matrix.yml:
seems outdated

greole commented on WindsorBody/basicSetupOBR/Allrun:
clean up

greole commented on WindsorBody/basicSetupOBR/Allclean:
clean up

greole commented on WindsorBody/basicSetupOBR/case.foam:
remove

greole commented on WindsorBody/basicSetupOBR/constant/turbulenceProperties:
Add RAS section and let OBR switch simulation type if needed

greole commented on WindsorBody/basicSetupOBR/constant/turbulenceProperties.LES:
remove

greole commented on WindsorBody/basicSetupOBR/constant/turbulenceProperties.RANS:
remove

greole commented on WindsorBody/basicSetupOBR/postProcess:
remove from merge branch

greole commented on WindsorBody/basicSetupOBR/system/controlDict:
remove and rely on OBR use function

greole commented on WindsorBody/basicSetupOBR/system/controlDict.DDES.avg:
remove

greole commented on WindsorBody/basicSetupOBR/system/controlDict.DDES.init:
copy to control.DDES

greole commented on WindsorBody/basicSetupOBR/system/createPatchDict.deleteIntern:
keep

greole commented on WindsorBody/basicSetupOBR/system/fvSchemes.pre:
remove

greole commented on WindsorBody/basicSetupOBR/system/fvSolution:
remove

greole commented on WindsorBody/basicSetupOBR/system/fvSolution.DDES:
keep

greole commented on WindsorBody/basicSetupOBR/system/include/caseDefinition:
keep and check if obr can adapt this?

greole commented on WindsorBody/benchmarkData/ExpData_LoughboroughUniv_VarneyEtAl/README_V2.txt:
Check if needed or should be moved to separate folder/repo

greole commented on WindsorBody/loadModules.sh:
leave out of merge branch

greole commented on common/plot_helpers.py:
revise this for separate PR

greole commented on common/postProcessLogs.py:
Move to data   repository?

greole committed:
clean up

greole committed:
clean up

greole committed:
structure yaml

greole committed:
keep logfiles out of this repo

greole committed:
use new uses notation

greole committed:
bump version
